compiler/src/dotty/tools/debug/ResolveReflectEval.scala

```diff
@@ -0,0 +1,346 @@
+      if field.owner.isTerm then
+        report.error(s"Cannot access local var ${field.name} in ${field.owner} as field", tree.srcPos)
+        ref(defn.Predef_undefined)
+      else
+        Apply(
+          Select(expressionThis, termName("setField")),
+          List(
+            qualifier,
+            Literal(Constant(JavaEncoding.encode(field.owner.asType))),
+            Literal(Constant(JavaEncoding.encode(field.name))),
+            value
+          )
+        )
+
+    def evaluateByName(function: Tree): Tree =
+      val castFunction = function.cast(defn.Function0.typeRef.appliedTo(defn.AnyType))
+      Apply(Select(castFunction, termName("apply")), List())
+
+    def callMethod(tree: Tree)(qualifier: Tree, method: TermSymbol, args: List[Tree]): Tree =
+      val methodType = method.info.asInstanceOf[MethodType]
+      val paramTypesNames = methodType.paramInfos.map(JavaEncoding.encode)
+      val paramTypesArray = JavaSeqLiteral(
+        paramTypesNames.map(t => Literal(Constant(t))),
+        TypeTree(ctx.definitions.StringType)
+      )
+
+      def unknownCapture(name: Name): Tree =
+        report.error(s"Unknown captured variable $name in $method", reflectEval.srcPos)
+        ref(defn.Predef_undefined)
+      val capturedArgs = methodType.paramNames.dropRight(args.size).map {
+        case name @ DerivedName(underlying, _) => capturedValue(tree)(method, underlying).getOrElse(unknownCapture(name))
+        case name => unknownCapture(name)
+      }
+
+      val erasedMethodInfo = atPhase(Phases.elimErasedValueTypePhase)(method.info).asInstanceOf[MethodType]
+      val unboxedArgs = erasedMethodInfo.paramInfos.takeRight(args.size).zip(args).map {
+        case (tpe: ErasedValueType, arg) => unboxValueClass(arg, tpe)
+        case (_, arg) => arg
+      }
+
+      val returnTypeName = JavaEncoding.encode(methodType.resType)
+      val methodName = JavaEncoding.encode(method.name)
+      val result = Apply(
+        Select(expressionThis, termName("callMethod")),
+        List(
+          qualifier,
+          Literal(Constant(JavaEncoding.encode(method.owner.asType))),
+          Literal(Constant(methodName)),
+          paramTypesArray,
+          Literal(Constant(returnTypeName)),
+          JavaSeqLiteral(capturedArgs ++ unboxedArgs, TypeTree(ctx.definitions.ObjectType))
+        )
+      )
+      erasedMethodInfo.resType match
+        case tpe: ErasedValueType => boxValueClass(tpe.tycon.typeSymbol.asClass, result)
+        case _ => result
+    end callMethod
+
+    def callConstructor(tree: Tree)(qualifier: Tree, ctr: TermSymbol, args: List[Tree]): Tree =
+      val methodType = ctr.info.asInstanceOf[MethodType]
+      val paramTypesNames = methodType.paramInfos.map(JavaEncoding.encode)
+      val clsName = JavaEncoding.encode(methodType.resType)
+
+      val capturedArgs =
+        methodType.paramNames.dropRight(args.size).map {
+          case outer if outer.toString == "$outer" => qualifier
```

Comment:
There should be an `nme.outer` or something. In general, avoid calling `toString` on names unless for debugging purposes.

Reply:
done